### DISCUSSION TITLE
Make telemetry opt-in, and clearly describe what data is collected

### DISCUSSION THREAD

**Discussion by zovt** [runelite/runelite#15698] [Ideas]
Hello, I updated my local branch this morning and discovered that there was a new `TelemetryClient` class. From a brief glance at the code, this looks be a new thing which is silently enabled by default, and the only way to tell which data is being collected (or that data is collected at all) is by reading the source code.

I would like to propose making this opt-in (potentially with a pop-up at first start) describing what data is collected, similar to the pop-ups that are currently present for some plugins which send data to 3rd parties.

Happy to help with any patches here, hopefully it shouldn't be too much work :) Thanks for your consideration and great client!

**Comment by Adam-** [comment]
I do understand your concern, however making the telemetry opt-in negates almost all usefulness of it to us. You are welcome to opt-out of it by passing `--disable-telemetry` to the client.

**Reply by zovt** [reply]
Fair enough, though I would urge you to make it abundantly clear that whenever the next release goes out that this new behavior is there lest people accuse you of malicious intent.

Thanks!
